app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -169,124 +139,64 @@ class MainActivity : BaseActivity(),
             removeFragment(SessionDetailsFragment.FRAGMENT_TAG)
         }
         initUserEngagement()
+        observeViewModel()
         onSessionAlarmNotificationTapped(intent)
     }
 
-    override fun onNewIntent(intent: Intent) {
-        super.onNewIntent(intent)
-        setIntent(intent)
-        onSessionAlarmNotificationTapped(intent)
-    }
-
-    private fun onSessionAlarmNotificationTapped(intent: Intent) {
-        val notificationId = intent.getIntExtra(BUNDLE_KEY_SESSION_ALARM_NOTIFICATION_ID, INVALID_NOTIFICATION_ID)
-        if (notificationId != INVALID_NOTIFICATION_ID) {
-            appRepository.deleteSessionAlarmNotificationId(notificationId)
+    private fun observeViewModel() {
+        viewModel.loadScheduleUiState.observe(this) {
+            updateUi(it)
```

Comment:
:x: Thank you for the pointer. From what I got from the article this would require to use `Flow` in the `Activity` or `Fragment`. I have decided not use `Flow` in the UI layer at this point. We definitely should remember this for later, though. :+1:

Reply:
I am curious - why you like to avoid Flows in the UI layer for now?

Reply:
Flow requires to use `repeatOnLifecycle` or others. Simply, I don't want to introduce these [very recently](https://developer.android.com/jetpack/androidx/releases/lifecycle#version_24_2) published functions at this point in time. Additionally, `lifecycle-livedata-ktx:2.4.0` requires SDK 31. We are currently using 30.